openedx/core/djangoapps/oauth_dispatch/views.py

```diff
@@ -75,6 +75,39 @@ def _get_client_id(self, request):
             return request.POST.get('client_id')
 
 
+def _get_token_type(request):
+    """
+    Get the token_type for the request.
+
+    - Respects the HTTP_X_TOKEN_TYPE header if the token_type parameter is not supplied.
+    - Adds `oauth_token_type` custom attribute for monitoring.
+    """
+    default_token_type = request.META.get('HTTP_X_TOKEN_TYPE', 'no_token_type_supplied')
+    token_type = request.POST.get('token_type', default_token_type).lower()
+    monitoring_utils.set_custom_attribute('oauth_token_type', token_type)
+    return token_type
+
+
+def _get_jwt_dict_from_opaque_access_token_dict(opaque_token_dict, oauth_adapter):
```

Comment:
**Note to Reviewer:** I've got second thoughts about introducing the term `opaque_token` into the method and argument names, because I think it is inconsistent, which may lead to confusion. I think I'll just leave it as `token_dict` for consistency, and leave the term "opaque" to comments where and when appropriate.

[question] Thoughts?

Reply:
I followed through on this change. Marking resolved.